scripts/core/editor3/react.tsx

```diff
@@ -31,6 +104,29 @@ export class Editor extends React.Component<any, any> {
         this.store = store;
     }
 
+    onChangeInterceptor(contentState: ContentState): void {
+        // ensure that onChange value is of the same type as received from the props
+        const rawState = convertToRaw(contentState);
+
+        this.props.onChange(rawState);
+    }
+
+    componentDidUpdate(prevProps: IProps) {
+        if (this.props.readOnly !== prevProps.readOnly) {
+            this.store.dispatch(setReadOnly(this.props.readOnly));
+        }
+
+        const rawStateFromStore = convertToRaw(this.store.getState().editorState.getCurrentContent());
+
+        if (isEqual(this.props.rawDraftContentState, rawStateFromStore) === false) {
+            // This component holds it's own state which is derived from props
+            // internal state is reloaded when it doesn't match with what's in the props
```

Comment:
in what situation can the editorState from props not be on sync with the actual editor state?

Reply:
When you are previewing item A in the knowledge base and switch to previewing item B.